logbook-spring-boot-starter/src/test/java/org/zalando/logbook/spring/ExcludeTest.java

```diff
@@ -0,0 +1,97 @@
+package org.zalando.logbook.spring;
+
+/*
+ * #%L
+ * Logbook: Spring
+ * %%
+ * Copyright (C) 2015 Zalando SE
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.zalando.logbook.Correlator;
+import org.zalando.logbook.Logbook;
+import org.zalando.logbook.MockRawHttpRequest;
+import org.zalando.logbook.RawHttpRequest;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static java.util.Optional.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ContextConfiguration
+@TestPropertySource(properties = "spring.config.name = exclude")
+public final class ExcludeTest extends AbstractTest {
+
+    @Configuration
+    public static class TestConfiguration {
+
+        @Bean
+        public Logger httpLogger() {
+            final Logger logger = mock(Logger.class);
+            when(logger.isTraceEnabled()).thenReturn(true);
+            return logger;
+        }
+        
+        @Bean
+        public Predicate<RawHttpRequest> requestPredicate() {
+            return request -> 
+                    !"/health".equals(request.getRequestUri());
```

Comment:
First of all, to summarize the problem (to double check if I understood it correctly): In order to implement the request predicate more easily, we want to extract the path from the URI?

In the example in the test, I'd just use a regex or the like (e.g. `"^/health($|\\?)"`) to match all URIs with that path.

More generally, I'd not introduce URL parsing directly into `RawHttpRequest` because this would mean that we need to cope with invalid URLs inside Logbook itself, which in turn could result in additional configuration options that influence the handling of those URLs. So, in my opinion, the implementor should cope with the parsing. Maybe no parsing is necessary at all, maybe a regex is enough, maybe some more complex logic needs to be in place. What could be offered by Logbook itself is some utility functionality for URL parsing that may be used by implementors, if appropriate. I experimented with a [fallback URL parsing mechanism](https://github.com/zalando/logbook/compare/master...twz123:leninet-obfuscate-urls#diff-4651f587bef3d5da34fd5047d667a09aR76) back in the days of #49 to address this. It just tries to parse an URL on a best-effort basis and never fails. Maybe this could be reworked to a real utility method/class that parses a String into an URI in a fault-tolerant way.